packages/@aws-cdk/aws-lambda/lib/edge-function.ts

```diff
@@ -0,0 +1,259 @@
+import * as path from 'path';
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as iam from '@aws-cdk/aws-iam';
+import * as ssm from '@aws-cdk/aws-ssm';
+import {
+  BootstraplessSynthesizer, Construct as CoreConstruct, ConstructNode,
+  CustomResource, CustomResourceProvider, CustomResourceProviderRuntime,
+  DefaultStackSynthesizer, IStackSynthesizer, Resource, Stack, StackProps, Stage, Token,
+} from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { Alias, AliasOptions } from './alias';
+import { EventInvokeConfigOptions } from './event-invoke-config';
+import { IEventSource } from './event-source';
+import { EventSourceMapping, EventSourceMappingOptions } from './event-source-mapping';
+import { Function, FunctionProps } from './function';
+import { IFunction } from './function-base';
+import { extractQualifierFromArn, IVersion } from './lambda-version';
+import { Permission } from './permission';
+
+/**
+ * Properties for creating a Lambda@Edge function
+ * @experimental
+ */
+export interface EdgeFunctionProps extends FunctionProps { }
+
+/**
+ * A Lambda@Edge function.
+ *
+ * Convenience resource for requesting a Lambda function in the 'us-east-1' region for use with Lambda@Edge.
+ * Implements several restrictions enforced by Lambda@Edge.
+ *
+ * @resource AWS::Lambda::Function
+ * @experimental
+ */
+export class EdgeFunction extends Resource implements IVersion {
+
```

Comment:
(related to the comment above)

I'm wondering how much further can we take this without tying ourselves into knots. (just brainstorming, no need yet to take this into implementation)

Let's say we cloned the `Function` construct into a new construct called `FunctionV2` (or `FunctionPlus`, `UberFunction`, etc.). However, instead of creating a `new CfnFunction()` in the constructor, it wraps it up inside a `Lazy`. The `Lazy`, when it's resolved, either creates a `CfnFunction` as a child construct of the current stack or one under a new 'us-east-1' stack, the decision determined by its usage.

Does this make everything very complicated?

Reply:
_(Capturing offline conversation)_

Probably a bit too side-effecty for my taste. However, willing to consider further with more user feedback.